bin/convert2vulyk.py

```diff
@@ -42,116 +83,173 @@ def convert_bsf_2_vulyk(text: str, bsf_markup: str) -> dict:
+        print(json.dumps(vulyk_obj, ensure_ascii=False, sort_keys=True))
 
-def _run_ner(txt: str) -> str:
-    """
-    Run ner pipeline on tokenized text using Stanza.
-    :param txt: tokenized text string
-    :return: string with annotations in brat standoff format
-    """
-    import stanza
-    logging.getLogger("stanza").setLevel(log.level)
 
-    stanza.download('uk')
+def tag(args):
+    if args.ner_framework == "stanza":
+        model = StanzaNER(args.ner_model)
+    elif args.ner_framework == "spacy":
+        model = SpacyNER(args.ner_model)
 
-    ner = stanza.Pipeline(lang='uk', processors='tokenize,mwt,ner', tokenize_pretokenized='true')
-    log.info("Processing text. It may take few seconds...")
-    doc = ner(txt)
+    for text in map(pathlib.Path, glob.glob(args.text_files)):
+        log.info(f"Found text file {text}, tagging it")
 
-    tok_i = 1
-    brat_str = ""
-    for ent in doc.ents:
-        brat_str += f'T{tok_i}\t{ent.type} {ent.start_char} {ent.end_char}\t{ent.text}\n'
-        tok_i += 1
-    return brat_str
+        token_list = tokenize_text(text.read_text())
+
+        # we have list<paragraphs> of list<sentences> of list<tokens>
+        paragraph = ["\n".join([" ".join(t) for t in sent]) for sent in token_list]
+        txt = "\n".join(paragraph)  # stanza bug does not allow for double new line symbol right now
+
+        markup = model.tag_text(txt)
+
+        vulyk_obj = convert_bsf_2_vulyk(txt, markup)
+
+        print(json.dumps(vulyk_obj, ensure_ascii=False, sort_keys=True))
 
 
 if __name__ == "__main__":
     logging.basicConfig()
 
     parser = argparse.ArgumentParser(
-        description='Convert text (tokenized or generic) to json file supported by Vulyk. '
-                    'Data file can be supplied via cmd line arguments '
-                    'or you can pipe data in and out of this script like:'
-                    '`cat file.txt | python3 convert2vulyk.py > save_to_file.json`')
-    parser.add_argument('text_file', nargs="?", type=argparse.FileType('r'), default=sys.stdin,
-                        help='Text file to process. If file path not provided, assumes stdin stream.'
-                             ' Must be tokenized if --brat_file is provided. '
-                             '*.ann for lang-uk data set')
-    parser.add_argument('--brat_annotation', type=argparse.FileType('r'), default=None,
-                        help='Path to file with Brat standoff markup')
-    parser.add_argument('-v', action='store_true', help='Print more logs (info)')
-    parser.add_argument('-vv', action='store_true', help='Print even more logs (debug)')
+        description="Convert tokenized pre-annotated text or tag generic text to json file supported by Vulyk. "
+        "Data file can be supplied via cmd line arguments "
+        "or you can pipe data in and out of this script like:"
+        "`cat file.txt | python3 convert2vulyk.py tag >> save_to_file.json`"
```

Comment:
With the variety of arguments, it would be great to have a few examples of how different modes are supposed to be called. Right now you can kinda figure it out but it is a puzzle.

Reply:
I'll take a look, if I can provide a more comprehensive text.